### PR TITLE
Add go.mod file

### PR DESCRIPTION
**Pull request by didrocks**
Add a go.mod for go module users.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module github.com/snapcore/go-gettext
+
+go 1.13
```
Comment (mvo5) [inline go.mod]: Silly question - what does "go 1.13" mean here? I looked at the docs and I could not quite find information what consequences this has :/

Comment (didrocks) [reply go.mod]: Sure, the doc is quite terse about it.

It currently doesn't force anything. It means that this has been tested in module mode successfully starting go 1.13 (a little bit like a minimum version selection,  but for go stdlib itself).

Upstream sketched up plans so that in the future, it will force (in module mode again) to have this minimum golang version to compile. My guess is that it will be helpful for them in particularly in the Go 2 transitions, if there are some breaking changes, to be able to compile some modules in Go 1 compatiblity mode and others in Go 2, as they told they want to be able to mix Go 1 and Go 2 code. But that's really just a guess :)

Setting it here to 1.13 is fine (even if this module is compatible with older Go stdlib version), because this is the first version with module out of experimental.
Basically, if you still compile with `GOPATH` (which I think snapd is doing on older release and still is today, even in 1.13), this has no impact.

**Comment by didrocks** [comment]
FYI, I have mirrored snapd using go-gettext in https://github.com/ubuntu/zsys/commit/b50a4046c84d0fa2d1a7a510104dd27a02410c89, stripping out some snapd dependencies and changing slightely the code to be more idiomatic.

I have drastically changed the tests though to be more idiomatic and follow other of my projects (including zsys): https://github.com/ubuntu/zsys/commit/639ad4f0d9b327b37ad54c0c82e1b353eb8c867a. Even if this turned out in more code, it covers more cases (12 vs 6) and adding a new testcase is a one-liner thanks to subtests.

File: go.mod (hunk 1 above)
Comment (mvo5) [reply go.mod]: Thanks for explaining this!

**Comment by didrocks** [comment]
Thanks for reviewing :)

**Comment by mvo5** [comment]
Thanks for the pointers btw to zsys btw, I don't have time right now but would love to see if we can share code in some common repo or something.
